Update docs in preparation for 1.0.0 release

jonahgraham opened the pull request:
Fixes #204

jonahgraham commented:
@ghentschke please let me know what additional docs you want me to add for 1.0 release.

ghentschke reviewed (commented):
LGTM

I've no other doc files to be added for the 1.0 release. Thank you.

jonahgraham commented:
Thanks @ghentschke  - this will be merged at the same time as we complete the release.

ghentschke commented:
Maybe add some new screenshots of the new Editor preference/properties page.

jonahgraham commented:
Sure thing. I'll do that!

jonahgraham commented:
> Maybe add some new screenshots of the new Editor preference/properties page.

Done.

I also did the following that achieves the steps in #209 

- Created changelog and releasing markdown files
- Update contribution guidelines to help better setup for development
- Added reference to CI jobs
- Update all screenshots in README to be consistent with current source
- Added a Version Support table to the readme
